pkg/controller/apiextensions/requirement/api.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package requirement
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// Error strings.
+const (
+	errCreateComposite   = "cannot create composite resource"
+	errUpdateRequirement = "cannot update resource requirement"
+	errUpdateComposite   = "cannot update composite resource"
+	errDeleteComposite   = "cannot delete composite resource"
+)
+
+// An APICompositeCreator creates resources by submitting them to a Kubernetes
+// API server.
+type APICompositeCreator struct {
+	client client.Client
+	typer  runtime.ObjectTyper
+}
+
+// NewAPICompositeCreator returns a new APICompositeCreator.
+func NewAPICompositeCreator(c client.Client, t runtime.ObjectTyper) *APICompositeCreator {
+	return &APICompositeCreator{client: c, typer: t}
+}
+
+// TODO(negz): We should render and patch a composite resource on each
```

Comment:
Do you plan to address this in this PR?

Reply:
No. I'd like to address this in a future PR.